Stock Ageing TypeError: ‘<’ not supported between instances of ‘int’ …

clarkejj opened the pull request:
…and ‘str’

Re this user report https://discuss.erpnext.com/t/stock-ageing-report-typeerror-not-supported-between-instances-of-int-and-str/62605

Notes: 

ASSUMPTION:  This error arises in current version-12 production?

WARNING: This proposed change has not been tested

<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/erpnext/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

clarkejj committed:
Stock Ageing TypeError: ‘<’ not supported between instances of ‘int’ and ‘str’

Proposed fix re this error report https://discuss.erpnext.com/t/stock-ageing-report-typeerror-not-supported-between-instances-of-int-and-str/62605

deepeshgarg007 commented on erpnext/stock/report/stock_ageing/stock_ageing.py:
```diff
@@ -180,14 +180,14 @@ def get_fifo_queue(filters, sle=None):
 				while qty_to_pop:
 					batch = fifo_queue[0] if fifo_queue else [0, None]
-					if 0 < batch[0] <= qty_to_pop:
+					if 0 < cint(batch[0]) <= qty_to_pop:
```
`flt` should be used instead of `cint` as qty can be fraction value as well

clarkejj committed:
Update stock_ageing.py

As reviewer Deepesh Garg advised thanks!  batch[0] can be int of float - see line 49 - so change proposed cint to flt

clarkejj reviewed (commented):
As Deepesh advised on review of my original proposed commit - change cint to flt since batch[0] can be int or float - see line 49

Thanks Deepesh!

marination commented on erpnext/stock/report/stock_ageing/stock_ageing.py:
```diff
@@ -180,14 +180,14 @@ def get_fifo_queue(filters, sle=None):
 					else:
 						# all from current batch
-						batch[0] -= qty_to_pop
+						flt(batch[0]) -= qty_to_pop
```
```suggestion
						batch[0] = flt(batch[0]) - qty_to_pop
```
Can't assign values to a function call so `flt(batch[0])` is problematic

clarkejj committed:
Thanks so much for your pointer Marica - that violation is clear and obvious to me in retrospect, but a good relearning experience in any case

clarkejj reviewed (commented):
Many thanks for your patient pointer Marica - that assignment violation is clear to me in retrospect, but a good relearning and change commit exercise in any case!

clarkejj committed:
Merge branch 'develop' into clarkejj-patch-1

clarkejj commented:
Hi Deepesh and Marica - so once this fix is ok with all checks passing, will you then push this to develop, version-12-hotfix and other backport expected branches where required?

deepeshgarg007 commented:
@Mergifyio backport version-12-hotfix

clarkejj commented:
Excellent thanks for the process learning coaching Deepesh and Marica

John

marination commented:
@clarkejj Can you please send these changes on version-12-hotfix as well ?
